srcs/app/http/HttpRequestHandler.cpp

```diff
@@ -22,92 +29,292 @@ bool		HttpRequestHandler::GetKeepAlive() const {
-	//           and the page exist
-	DefaultErrorPage_(error_code);
+void	HttpRequestHandler::RequestError_(const Location *location,
+											const std::size_t error_code) {
+	const CommonConfig &cfg = GetCommonConfig(location);
+	CommonConfig::ErrorPagesMap::const_iterator it =
+											cfg.error_pages.find(error_code);
+	if (it != cfg.error_pages.end()) {
+		const std::string error_page = cfg.root + it->second;
+		ServeFile_(location, error_page);
+		return;
+	}
+	DefaultStatusResponse_(error_code);
 }
 
-void	HttpRequestHandler::DoGet_(const HttpRequest &request) {
-	// TODO(any) Implement GET
-	(void)request;
+void	HttpRequestHandler::PathError_(const Location *location) {
+	if (errno == ENOENT || errno == ENOTDIR) {
+		RequestError_(location, 404);
+	} else if (errno == EACCES) {
+		RequestError_(location, 403);
+	} else {
+		RequestError_(location, 500);
+	}
+}
+
+bool	HttpRequestHandler::TryAddDirectoryContent_(std::stringstream *body,
+											const Location *location,
+											const std::string &full_path) {
+	DIR *dir = opendir(full_path.c_str());
+	if (dir == NULL) {
+		PathError_(location);
+	}
+	struct dirent *entry;
+	while ((entry = readdir(dir)) != NULL) {
+		std::string name = entry->d_name;
+		if (name.rfind(".", 0) == 0) {
+			continue;
+		}
+		const std::string full_path_name = full_path + "/" + name;
+		if (IsDirectory_(full_path_name)) {
+				name.append("/");
+		}
+		*body << "<a href=\"" << name << "\">" << name << "</a>\n";
+	}
+	closedir(dir);
+	return true;
+}
+
+std::string	HttpRequestHandler::GetFullPath_(const Location *location,
+										const std::string &request_path) const {
+	if (!location) {
+		return server_config_.common.root + request_path;
+	}
+	return location->common.root + request_path;
+}
+
+void	HttpRequestHandler::ListDirectory_(const Location *location,
+											const std::string &request_path) {
+	std::stringstream body;
+	body << "<html>\n" <<
+		"<head><title>Index of " << request_path << "</title></head>\n" <<
+		"<body>\n" <<
+		"<h1>Index of " << request_path <<
+		"</h1><hr><pre><a href=\"../\">../</a>\n";
+	const std::string full_path = GetFullPath_(location, request_path);
+	if (!TryAddDirectoryContent_(&body, location, full_path)) {
+		return;
+	}
+	body << "</pre><hr></body>\n" <<
+		"</html>\n";
 	HttpResponse	response(200);
-	std::string		body = "Responding to a GET request\n";
+	response.AddHeader("Content-Type", "text/html");
+	response.SetBody(body.str());
+	AddCommonHeaders_(&response);
+	raw_response_ = response.CreateResponseString();
+}
 
-	response.AddHeader("Content-Type", "text/plain");
+const CommonConfig &
+HttpRequestHandler::GetCommonConfig(const Location *location) const {
+	if (location) {
+		return location->common;
+	}
+	return server_config_.common;
+}
+
+bool	HttpRequestHandler::HasAcceptedFormat_(const Location *location,
+												const HttpRequest &request) {
+	if (location && !location->limit_except.empty()) {
```

Comment:
I don't want us to go in circles, so, for me, this is MR can be said good to go, but just so you know my opinion on these choices, here you have it:

You keep having to check for Location to not be NULL.
Wouldn't we have better encapsulation if Location was always constructed, and had an additional field (a bool), that expressed whether the object is valid or not, and all setters and getters  (that should be in the Location class) reacted to it?
Like that, you could not forget to check every time, and generally the code would be more maintainable.